LearnR-course.Rmd

```diff
@@ -444,30 +444,13 @@ tail(gender_data)
 ```
 
 ``` {r headtail-check}
+ex() %>% check_function("head") %>% check_result() %>% check_equal()
 state <- ex()
 check_function(state, "head") %>% check_arg("x") %>% check_equal()
 
-# This section of the code check gives an incorrect answer to the head section.
-#check_output(state,
-#  '1    Afghanistan   4\\.95450  19961015094[[:space:]]+     161\\.1380',
-#  missing_msg='Did you display the value for "head(gender_data)"?')
-
-# This gives a correct answer, but adding more to the regex arg causes an incorrect message.
-check_output(state,
-             'Afghanistan',
-             missing_msg='Did you display the value for "head(gender_data)"?')
-
+ex() %>% check_function("tail") %>% check_result() %>% check_equal()
```

Comment:
This looks good. It would be nice if we had a test that validates this throws an error if the user provides `head()` with an explicit number of records that isn't 6. However, as hooking into the inside of LearnR seems tricky, I think we'll live with you having manually verified this.

Please could you capture this learning in an issue, describing the original problem and (i) the solution and (ii) the things that looked like they might have worked but didn't.